src/Raven.Server/Utils/SecurityUtils.cs

```diff
@@ -0,0 +1,61 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Net;
+using System.Net.Sockets;
+using System.Text;
+using Raven.Server.Config;
+using Raven.Server.Config.Categories;
+
+namespace Raven.Server.Utils
+{
+    public static class SecurityUtils
+    {
+        public static bool IsUnsecuredAccessAllowedForAddress(UnsecuredAccessAddressRange addressRange, IPAddress ipAddress)
+        {
+            switch (ipAddress.AddressFamily)
+            {
+                case AddressFamily.InterNetwork:
+                    return IsUnsecuredAccessAllowedForIPv4Address(addressRange, ipAddress);
+                case AddressFamily.InterNetworkV6:
+                    return IsUnsecuredAccessAllowedForIPv6Address(addressRange, ipAddress);
+                default:
+                    throw new NotSupportedException($"Not supported address family: { ipAddress.AddressFamily }");
+            }
+        }
+
+        private static bool IsUnsecuredAccessAllowedForIPv4Address(
+            UnsecuredAccessAddressRange addressRange, IPAddress address)
+        {
+            var octets = address.GetAddressBytes();
+
+            return (AddressRangeIncludes(addressRange, UnsecuredAccessAddressRange.Local) && octets[0] == 127)
+                || (AddressRangeIncludes(addressRange, UnsecuredAccessAddressRange.PrivateNetwork) && (octets[0] == 10 || octets[0] == 192 || octets[0] == 172))
```

Comment:
Checking just `192` is not valid, you need to check `192.168`, same for `172. 
See https://www.arin.net/knowledge/address_filters.html